nixos/modules/security/sudo.nix

```diff
@@ -61,19 +168,35 @@ in
 
   config = mkIf cfg.enable {
 
+    security.sudo.extraRules = [
+      { groups = [ "wheel" ];
+        commands = [ { command = "ALL"; options = (if cfg.wheelNeedsPassword then [ "NOPASSWD" ] else [ ]); } ];
+      }
+    ];
+
     security.sudo.configFile =
       ''
         # Don't edit this file. Set the NixOS options ‘security.sudo.configFile’
-        # or ‘security.sudo.extraConfig’ instead.
+        # or ‘security.sudo.extraRules’ instead.
 
         # Keep SSH_AUTH_SOCK so that pam_ssh_agent_auth.so can do its magic.
         Defaults env_keep+=SSH_AUTH_SOCK
 
         # "root" is allowed to do anything.
         root        ALL=(ALL:ALL) SETENV: ALL
 
-        # Users in the "wheel" group can do anything.
-        %wheel      ALL=(ALL:ALL) ${if cfg.wheelNeedsPassword then "" else "NOPASSWD: ALL, "}SETENV: ALL
```

Comment:
It now generates this line:

```
wheel  ALL=(ALL:ALL) : ALL
```

which misses the `SETENV` prefix and is therefore invalid:

```
parse error in /nix/store/76iail0ygnc0f81xp4lr42jg2ac1vy92-sudoers-in near line 11
```

You can test this by running: `nix-build nixos/tests/misc.nix`

Reply:
Oh, I actually didn't test it with a command in the `options` syntax but with an empty list of options. Now it should work as expected and I even pass the `SETENV` option.

Prior to this change, it generated
```
%wheel      ALL=(ALL:ALL) NOPASSWD: ALL, SETENV: ALL
```

Now, the line looks like
```
%wheel  ALL=(ALL:ALL)   NOPASSWD:SETENV: ALL
```
which should have the exact same effect.

All the tests defined in `nixos/tests/misc.nix` are succeeding too.